yt_dlp/extractor/theta.py

```diff
@@ -0,0 +1,45 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import try_get
+
+
+class ThetaIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?theta\.tv/(?P<id>[a-z0-9]+)'
+    _TESTS = [{
+        'url': 'https://www.theta.tv/davirus',
+        'info_dict': {
+            'id': 'DaVirus',
+            'ext': 'mp4',
+            'title': 'I choose you - My Community is King -👀 - YO HABLO ESPANOL - CODE DAVIRUS',
+            'thumbnail': r're:https://live-thumbnails-prod-theta-tv\.imgix\.net/thumbnail/.+\.jpg',
+        }
+    }, {
+        'url': 'https://www.theta.tv/mst3k',
+        'info_dict': {
+            'id': 'MST3K',
+            'ext': 'mp4',
+            'title': 'Mystery Science Theatre 3000 24/7 Powered by the THETA Network.',
+            'thumbnail': r're:https://user-prod-theta-tv\.imgix\.net/.+\.jpg',
+        }
+    }]
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+        info = self._download_json(f'https://api.theta.tv/v1/channel?alias={channel_id}', channel_id)['body']
+
+        for data in try_get(info, lambda x: x['live_stream']['video_urls']):
+            if (data.get('type') != 'embed') and ((data.get('resolution') == 'master') or (data.get('resolution') == 'source')):
+                m3u8_playlist = data.get('url')
```

Comment:
```suggestion
        m3u8_playlist = next(
            data['url']
            if data.get('type') != 'embed' and data.get('resolution') in ('master', 'source')
            for data in info['live_stream']['video_urls'])
```

Reply:
This one fails with syntax error

Reply:
Sorry, that's my mistake. The `if` comes after `for`

```suggestion
        m3u8_playlist = next(
            data['url'] for data in info['live_stream']['video_urls']
            if data.get('type') != 'embed' and data.get('resolution') in ('master', 'source'))
```